### PR TITLE
fixed incorrect reading of day and month

### PR DESCRIPTION
**Pull request by haydenlj**
## Description

corrected day and month being read in from file
### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #<issue_number>

## Acceptance Criteria (Definition of Done)

bug is fixed and month and day are correct for 2 digit month and day

### CODE REVIEW

**Review by BenjaminRuston — COMMENTED** [review]
do we need to add an exception to ignore `MetaData/windHeightAssignMethod`

`169:   Variable MetaData/windHeightAssignMethod
169:    ERROR: Variable 'MetaData/windHeightAssignMethod' needs units, but the 'units' attribute does not exist in the file.`


and now am getting an missing units error for `PreQC/eastward_wind` and `PreQC/northward_wind`

**Comment by gthompsnJCSDA** [comment]
@BenjaminRuston   I think you have some weird combination of branches perhaps.  There are no units for any PreQC variable and the windHeightAssignMethod should not have units either as I recall.  Most importantly, the ci-testing is passing which says to me that things are done correctly.  What you have in your local branch or compilation must have something that contains a mixture of things from sprint-ioda-converters.

So I am just hitting APPROVE button for this PR.

**Review by gthompsnJCSDA — APPROVED** [review]
Is this really how python works with indexes to get a substring?  Very strange that it would use a beginning number like zero, but 4 (for the year) indicates that 4 is not inclusive rather exclusive.  So one could have said:
```
year = int(row['day'][0:4]
month = int(row['day'][4:6]
day = int(row['day'][6:9]
```
which seems awfully stupid since 9 is beyond range of 8 characters.  Fortran is smarter in this context with "inclusive" indexes.

Regardless, I believe this is needed and fixes a bug.

**Comment by haydenlj** [comment]
> Is this really how python works with indexes to get a substring? Very strange that it would use a beginning number like zero, but 4 (for the year) indicates that 4 is not inclusive rather exclusive. So one could have said:
> 
> ```
> year = int(row['day'][0:4]
> month = int(row['day'][4:6]
> day = int(row['day'][6:9]
> ```
> 
> which seems awfully stupid since 9 is beyond range of 8 characters. Fortran is smarter in this context with "inclusive" indexes.
> 
> Regardless, I believe this is needed and fixes a bug.

Unfortunately yes, Python indexing is inclusive of the first index given and exclusive of the last. day = int(row['day'][6:9] would also work and give the same output I believe

**Comment by CoryMartin-NOAA** [comment]
@gthompsnJCSDA It should probably be `int(row['day'][6:8])` or just `int(row['day'][6:])`. You could do `int(row['day'][6:9999999999])` and it would give the same result.

**Comment by BenjaminRuston** [comment]
@gthompsnJCSDA  your second suggestion is the one seen the most often

`int(row['day'][6:])`

**Review by BenjaminRuston — APPROVED** [review]

File: src/conventional/amv_ssec_ascii2ioda.py [hunk 1]
```diff
@@ -180,8 +180,8 @@ def read_file(file_name, data):
         for row in reader:
             try:
                 year = int(row['day'][0:4])
-                month = int(row['day'][5:6])
-                day = int(row['day'][7:8])
+                month = int(row['day'][4:6])
+                day = int(row['day'][-2:])
                 hour = int(row['hms'][0:1])
```
[the remark this answers is not in the document]
Comment (gthompsnJCSDA) [reply src/conventional/amv_ssec_ascii2ioda.py]: Good catch Ben!  Thanks.

**Pull request merged**